src/applications/disability-benefits/all-claims/utils.jsx

```diff
@@ -190,6 +190,46 @@ export function transformProviderFacilities(providerFacilities) {
   }));
 }
 
+/**
+ * This is mostly copied from us-forms' own stringifyFormReplacer, but with
+ * the incomplete / empty address check removed, since we don't need this
+ * for any of the 3 addresses (mailing, forwarding, treatment facility) in our
+ * form. Leaving it in breaks treatment facility addresses because by design
+ * they don't have street / line 1 addresses, so would get incorrectly filtered
+ * out. Trivia: this check is also gone in the latest us-forms replacer.
+ */
+export function customReplacer(key, value) {
+  // clean up empty objects, which we have no reason to send
+  if (typeof value === 'object') {
```

Comment:
So this is a shallow checker. If we have something like
```js
{
  parent: {
    child: {}
  }
}
```
what we'd be left with at the end would be
```js
{
  parent: {}
}
```
To solve this, I wrote a `removeDeeplyEmptyObjects()` which we now use in the `transform` function. We could pull that out of there to put it in here in place of this shallow remover.

Reply:
I like that idea generally, but am concerned with deviating from the common replacer more than necessary. I think the helper you've got might be a nice addition to our version of `us-forms-system`, since it would seem to be a good idea across all forms. Thoughts?